### PR TITLE
Added the option to change the icon file extension from png to other …

### PR DESCRIPTION
**Pull request by plyoung**
Added the option to change the icon file extension from png to other like tres or svg

### CODE REVIEW

File: addons/controller_icons/Settings.gd [hunk 1]
```diff
@@ -45,6 +45,9 @@ enum Devices {
 ## Custom generic joystick mapper script
 @export var custom_mapper : Script
 
+## Custom icon file extension
+@export var file_extension : String = "png"
```
Comment (rsubtil) [inline addons/controller_icons/Settings.gd:49]: Could you also rename this to `custom_file_extension` to further denote this?
```suggestion
@export var custom_file_extension : String = "png"
```

**Review by rsubtil — CHANGES_REQUESTED** [review]
Thank you for your contribution! There's only a minor point that I'd like to change regarding how this works.

All custom properties work in the basis of having no data whatsoever if the user doesn't specify it explicitly. So in general, the code tests if these custom properties have any content, and it not, fallback to a default value, such as in `custom_label_settings`, `custom_mapper` and `custom_asset_dir`:

https://github.com/rsubtil/controller_icons/blob/bd58e50c8b39dafd8f23f0a82da2c380ea713c49/addons/controller_icons/objects/ControllerIconTexture.gd#L135-L139
https://github.com/rsubtil/controller_icons/blob/bd58e50c8b39dafd8f23f0a82da2c380ea713c49/addons/controller_icons/ControllerIcons.gd#L102-L107
https://github.com/rsubtil/controller_icons/blob/bd58e50c8b39dafd8f23f0a82da2c380ea713c49/addons/controller_icons/ControllerIcons.gd#L197-L202

So in that vein, could you change the `file_extension` to follow this same pattern? Essentially, have the default `png` extension and code, and changing it if this settings is not empty:

```gdscript
# In ControllerIcons.gd
var base_extension := "png"

...
func _ready():
    ...
    if _settings.custom_file_extension:
        base_extension = _settings.custom_file_extension
    ...
```

**Comment by plyoung** [comment]
I've made the changes. Hope it is right. I've not worked with pull requests before.

**Comment by rsubtil** [comment]
> I've made the changes. Hope it is right. I've not worked with pull requests before.

It's alright! There were some minor issues left, but I went ahead and added a commit to tackle them:

- Renamed to `_base_extension` since it is a private variable
- Move `var _base_extension` elsewhere to be more inline with other variables
- Regenerate `settings.tres` as it was still using the old `file_extension` name

**Review by rsubtil — APPROVED** [review]

**Comment by rsubtil** [comment]
Thank you so much for your contribution! :tada:

**Pull request merged**